network/src/validator_network/consensus.rs

```diff
@@ -100,25 +100,13 @@ impl ConsensusNetworkSender {
     pub async fn request_block(
```

Comment:
rename `request_block` to `send_rpc`?

Reply:
annoying that we send and receive ConsensusMsg here and not the specific enum variants...

Reply:
the original intent at least is for the `request_block` API to take in a `RequestBlock`, return a `ResponseBlock`, and handle lowering into a `ConsensusMsg` internally. if we're just going to take a blanket `ConsensusMsg` in and output a `ConsensusMsg`, we might as well just rename this to `send_rpc` or `unary_rpc` or whatever.

Reply:
I think this is a consistent theme in these files. I bet we could build a trait + a default implementation that takes in a generic that specifies the protocol id. That would likely eliminate a lot of the code and tests, but wanted to sync with you folks.

Enum separation is so that we don't leak consensus info here. I'm curious what the right interface should be... should it be as generic as a Vec<u8> / &[u8] or something a bit more indicative of its function.